File: frontend/lib/models/proposal.dart
import 'package:frontend/models/flight.dart';
import 'package:frontend/models/user.dart';

class Proposal {
  int id;
  String departureTime;
  String description;
  int availableSeats;
  Flight flight;

  Proposal({
    required this.id,
    required this.departureTime,
    required this.description,
    required this.availableSeats,
    required this.flight,
  });

  factory Proposal.fromJson(Map<String, dynamic> json) {
    return Proposal(
      id: json['id'],
      departureTime: json['departure_time'],
      description: json['description'],
      availableSeats: json['available_seats'],
      flight: Flight.fromJson(json['flight']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'departure_time': description,
      'description': description,
      'available_seats': availableSeats,
      'flight': flight.toJson(),
    };
  }
}

File: backend/models/proposal.go
package models

import (
	"time"
)

type Proposal struct {
	ID             int       `json:"id" gorm:"primaryKey;autoIncrement:true"`
	DepartureTime  time.Time `json:"departure_time" gorm:"not null"`
	Description    string    `json:"description" gorm:"not null"`
	AvailableSeats int       `json:"available_seats" gorm:"not null"`
	FlightID       int       `json:"flight_id" gorm:"not null"`
	Flight         Flight    `json:"flight" gorm:"foreignkey:FlightID"`
}
